smoketest: T7996: fix hardcoded "vrf red" search path in FRR config

c-po opened the pull request:
<!-- All PR should follow this template to allow a clean and transparent review -->
<!-- Text placed between these delimiters is considered a comment and is not rendered -->

## Change summary
<!--- Provide a general summary of your changes in the Title above -->

Commit 186900f7165b2 (`smoketest: T7927: test DHCP route preservation`) which was added to validate a bugfix for DHCP default routes added it's own little regression. Tests defined a VRF named red15, but when reading in the FRR configuration we have had a hardcoded search for `vrf red` instead. 

This has been fixed to re-use the defined VRF variable name we perform our test on.

The original VRF DHCP test case, introduced in commit 186900f7165b2 (`smoketest: T7927: test DHCP route preservation`), used a fixed delay of 8 seconds to wait for the DHCP server and client to initialize. This approach was unreliable and could cause unnecessary test delays.

Since commit 354517677f (`wlb: T7977: Updated smoketest to validate nft vmap weight buckets`), the `vyos.utils.misc.wait_for()` helper provides a more efficient busy-wait loop with an early-exit condition. This test case now uses `wait_for()` to detect when the DHCP-assigned default route becomes available, eliminating the fixed sleep and reducing test runtime.

## Types of changes
<!---
What types of changes does your code introduce? Put an 'x' in all the boxes that apply.
NOTE: Markdown requires no leading or trailing whitespace inside the [ ] for checking
the box, please use [x]
-->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)
<!-- optional: Link to related other tasks on Phabricator. -->
* https://vyos.dev/T7996

## Related PR(s)
<!-- Link here any PRs in other repositories that are required by this PR -->

## How to test / Smoketest result

Local smoketests runs of `make testc`, `make test-no-interfaces-no-vpp`  and `make test-interfaces` completed successfully

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--- The entire development process is outlined here: https://docs.vyos.io/en/latest/contributing/development.html -->
- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

c-po committed:
smoketest: T7996: vrf CLI path is always deleted in tearDown()

c-po committed:
smoketest: T7996: fix hardcoded "vrf red" search path in FRR config

Commit 186900f7165b2 ("smoketest: T7927: test DHCP route preservation") which
was added to validate a bugfix for DHCP default routes added it's own
little regression. Tests defined a VRF named red15, but when reading in the FRR
configuration we have had a hardcoded search for "vrf red" instead.

This has been fixed to re-use the defined VRF variable name we perform our test
on.

c-po committed:
smoketest: T7996: replace hardcoded sleep(8) in VRF test with wait_for()

The original VRF DHCP test case, introduced in commit 186900f7165b2
("smoketest: T7927: test DHCP route preservation"), used a fixed delay of
8 seconds to wait for the DHCP server and client to initialize. This approach
was unreliable and could cause unnecessary test delays.

Since commit 354517677f ("wlb: T7977: Updated smoketest to validate nft vmap
weight buckets"), the "vyos.utils.misc.wait_for()" helper provides a more
efficient busy-wait loop with an early-exit condition. This test case now uses
wait_for() to detect when the DHCP-assigned default route becomes available,
eliminating the fixed sleep and reducing test runtime.

sever-sever reviewed (approved):
Fix VRF smoketest
Dehardcode VRF
Use `wait_for`

c-po committed:
smoketest: T7966: perform full regex match with ^ and $

Commit 9775bb49e ("frr: T7664: drop BASH/SED implementation in smoketest
getFRRconfig()") changed how we searched strings using a regex. In the
past we searched at the beginning of a line ^ till the end $. THis logic
was dropped in commit 9775bb49e marking some tests failing as they matched,
when they should not match.

Example: getFRRconfig('vrf red') showed output for both VRF red and red15 as
both started with "red". This has been fixed.

c-po committed:
smoketest: T7996: drop frr-reload.py implementation when reading running config

Commit 9775bb49e ("frr: T7664: drop BASH/SED implementation in smoketest
getFRRconfig()") dropped a custom BASH/SED based implementation of how we
interacted with FRR to make use of the utility helpers from FRR team.

This changed added some arbitray warnings when executing about unclosed
files handles. ResourceWarning: unclosed file <_io.BufferedReader name=5>

Instead of importing the frr-reload Python module, we simply call popen() by
ourself.

jestabro reviewed (approved):
Several subtle fixes for logic of smoketests/warnings: all vrf smoketests now pass in local test.
